XLS-0065d-single-asset-vault/README.md

```diff
@@ -0,0 +1,653 @@
+  - `MPToken(MPTokenIssuanceID, AccountID).MPTAmount` < `Amount` (insufficient balance).
+
+- The `Asset` is an `IOU`:
+  - The `lsfGlobalFreeze` flag is set on the issuing account (the asset is frozen).
+  - The `lsfHighFreeze` or `lsfLowFreeze` flag is set on the `RippleState` object between the Asset `Issuer` and the depositor.
+  - The `RippleState` object `Balance` < `Amount` (insufficient balance).
+
+##### 3.2.1.2 State Changes
+
+If no `MPToken` object exists for the depositor, create one. For object details, see [3.4.2 `MPToken`](#3.4.2-MPToken).
+
+- Increase the `MPTAmount` field of the share `MPToken` object of the `Account` by $\Delta_{share}$.
+- Increase the `OutstandingAmount` field of the share `MPTokenIssuance` object by $\Delta_{share}$.
+- Increase the `SharesTotal` of the `Vault` object by $\Delta_{share}$.
+
+- Increase the `SharesTotal` by $\Delta_{share}$.
+- Increase the `AssetsTotal` and `AssetsAvailable` by `Amount`.
+
+- If the `Vault.Asset` is `XRP`:
+
+  - Increase the `Balance` field of _pseudo-account_ `AccountRoot` by `Amount`.
+  - Decrease the `Balance` field of the depositor `AccountRoot` by `Amount`.
+
+- If the `Vault.Asset` is an `IOU`:
+
+  - Increase the `RippleState` balance between the _pseudo-account_ `AccountRoot` and the `Issuer` `AccountRoot` by `Amount`.
+  - Decrease the `RippleState` balance between the depositor `AccountRoot` and the `Issuer` `AccountRoot` by `Amount`.
+
+- If the `Vault.Asset` is an `MPT`:
+
+  - Increase the `MPToken.MPTAmount` by `Amount` of the _pseudo-account_ `MPToken` object for the `Vault.Asset`.
+  - Decrease the `MPToken.MPTAmount` by `Amount` of the depositor `MPToken` object for the `Vault.Asset`.
+
+- Increase the `AssetsTotal` and `AssetsAvailable` of the `Vault` by $\Delta_{asset}$.
+
+##### 3.2.1.3 Invariants
+
+**TBD**
+
+[**Return to Index**](#index)
+
+#### 3.2.2 `VaultWithdraw` transaction
+
+The `VaultWithdraw` transaction withdraws assets in exchange for the vault's shares.
+
+| Field Name        |     Required?      | JSON Type | Internal Type | Default Value | Description                                                             |
+| ----------------- | :----------------: | :-------: | :-----------: | :-----------: | :---------------------------------------------------------------------- |
+| `TransactionType` | :heavy_check_mark: | `string`  |   `UINT16`    |     `62`      | Transaction type.                                                       |
+| `VaultID`         | :heavy_check_mark: | `string`  |   `HASH256`   |     `N/A`     | The ID of the vault from which assets are withdrawn.                    |
+| `AssetAmount`     | :heavy_check_mark: | `number`  |   `NUMBER`    |       0       | The exact amount of Vault asset to withdraw.                            |
+| `ShareAmount`     | :heavy_check_mark: | `number`  |   `NUMBER`    |       0       | The exact amount of shares the account is willing to redeem.            |
```

Comment:
Just have one field, `Amount`. It includes an issue, which can match either the vault asset, for withdraw, or vault share, for redeem.

Reply:
Updated.